rotkehlchen/tests/unit/test_eth2.py

```diff
@@ -442,16 +460,53 @@ def test_validator_daily_stats_with_genesis_event(
 
 @pytest.mark.skipif('CI' in os.environ, reason='do not run this in CI as it spams')
 @pytest.mark.parametrize('default_mock_price_value', [FVal(1.55)])
-def test_scrape_genesis_validator_stats_all(price_historian):  # pylint: disable=unused-argument
+def test_scrape_genesis_validator_stats_all(
+        database: DBHandler,
+        messages_aggregator: MessagesAggregator,
+        price_historian,
+):  # pylint: disable=unused-argument
     """A test to see that the scraper of daily stats goes to the very first day of
     genesis for old validators despite UI actually showing pages"""
-    stats = scrape_validator_daily_stats(
+    beaconchain = BeaconChain(database, messages_aggregator)
+    stats = beaconchain.query_validator_daily_stats(
         validator_index=1,
         last_known_timestamp=Timestamp(0),
         exit_ts=None,
     )
     assert len(stats) >= 861
-    assert stats[-1].timestamp == 1606694400
+    assert stats[-1].timestamp == 1606694423
+
+
+@pytest.mark.freeze_time('2024-06-20 19:55:00 GMT')
+@pytest.mark.parametrize('network_mocking', [False])
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+def test_querying_ofter_stats(
+        database: DBHandler,
+        messages_aggregator: MessagesAggregator,
+        price_historian: 'PriceHistorian',
+        eth2: 'Eth2',
+):  # pylint: disable=unused-argument
+    """Test that querying twice the daily stats doesn't trigger a network
+    request the second time"""
+    eth2.add_validator(validator_index=1118010, public_key=None, ownership_proportion=ONE)
+    with eth2.database.conn.cursor() as cursor:
+        stats = eth2.get_validator_daily_stats(
+            cursor=cursor,
+            filter_query=Eth2DailyStatsFilterQuery.make(),
+            only_cache=False,
+        )
+        assert len(stats) > 100
+
+        with patch(
+            'rotkehlchen.chain.ethereum.modules.eth2.beacon.BeaconInquirer.query_validator_daily_stats',
+            wraps=eth2.beacon_inquirer.query_validator_daily_stats,
+        ) as scraper:
```

Comment:
This is no longer scraping.

```suggestion
        ) as patched_query:
```